### PR TITLE
modules: tf-m: split configuration into multiple Kconfig files

### PR DESCRIPTION
**Pull request by ioannisg**
Split the Kconfig-based TF-M configuration into multiple Kconfig files, for readability and to be able to easily override whole Kconfig files in downstream projects.

### CODE REVIEW

File: modules/trusted-firmware-m/Kconfig [hunk 1]
```diff
@@ -1,377 +1,12 @@
...
-	bool
-	help
-		This option instructs west flash to program the
-		combined (merged) binary consisting of the TF-M
-		Secure firmware image, optionally, the BL2 image
-		(if building with TFM_BL2 is enabled), and the
-		Non-Secure application firmware.
-
-endif # BUILD_WITH_TFM
+rsource "Kconfig.tfm.partitions"
```
Comment (hakonfam) [inline modules/trusted-firmware-m/Kconfig:10]: Both these files are "covered" internally by `if BUILD_WITH_TFM` - is it still necessary to have that check here?

Comment (ioannisg) [reply modules/trusted-firmware-m/Kconfig:8]: Removed, @hakonfam , thanks

**Review by hakonfam — APPROVED** [review]
My initial commit, so a bit biased :)

**Comment by ioannisg** [comment]
@microbuilder @oyvindronningstad @lairdjm, quick look?

**Comment by oyvindronningstad** [comment]
Since we lost a change downstream, I just want to ask if this we are sure there is unintended change?

**Review by oyvindronningstad — APPROVED** [review]

**Comment by hakonfam** [comment]
> Since we lost a change downstream, I just want to ask if this we are sure there is unintended change?

Should not be a problem since its just a duplicate check, I guess its enough that it will be resolved on the next upmerge.

File: modules/trusted-firmware-m/Kconfig.tfm [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+	int "Granularity of FW updates of TFM and app"
+	range 1 2
+	default 2
+	help
+	  How many images the bootloader sees when it looks at TFM and the app.
+	  When this is 1, the S and NS are considered as 1 image and must be
+	  updated in one atomic operation. When this is 2, they are split and
+	  can be updated independently if dependency requirements are met.
+
+
```
Comment (lairdjm) [inline modules/trusted-firmware-m/Kconfig.tfm:159]: Excess space

**Review by nordicjm — APPROVED** [review]

**Pull request merged**